hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManager.java

```diff
@@ -3512,7 +3512,11 @@ private Block addStoredBlock(final BlockInfo block,
     int numUsableReplicas = num.liveReplicas() +
         num.decommissioning() + num.liveEnteringMaintenanceReplicas();
 
-    if(storedBlock.getBlockUCState() == BlockUCState.COMMITTED &&
+
+    // if block is still under construction, then done for now
+    if (!storedBlock.isCompleteOrCommitted()) {
```

Comment:
Why do we move this block here?
BTW we can leave it as a single if with a return.

Reply:
I felt quite confused with the original structure since the early return was put after the statements it is trying to avoid..
I can make it a single return no big deal.